napari/layers/points/points.py

```diff
@@ -273,10 +276,17 @@ def __init__(
 
         # Save the properties
         if properties is None:
-            properties = {}
-        elif not isinstance(properties, dict):
+            self._properties = {}
+        elif len(data) > 0:
             properties = dataframe_to_properties(properties)
-        self._properties = self._validate_properties(properties)
+            self._properties = self._validate_properties(properties)
+            self._property_choices = {
+                k: np.unique(v) for k, v in properties.items()
+            }
+        elif len(data) == 0:
+            self._property_choices = properties
+            empty_properties = {k: np.empty(0) for k in properties}
+            self._properties = empty_properties
```

Comment:
I’d like this to be a tiny bit more elaborate. Specifically, properties can be a choice among objects, where np.unique makes sense, but it can also be a float or an int. I’d like to support the following syntax:

```python
layer.add_points(properties={‘class’: [‘a’, ‘b’], ‘value’: float})
```

The empty arrays should then have the correct dtype for the given property choices.

Reply:
I agree that we should make sure we are initializing the property arrays with the correct dtype. However, I think the user always needs to provide at least one value so that when a point gets added to the empty layer, it is given a value for all properties. I propose that we continue requiring that properties is a dictionary with values that are numpy arrays and that we use the `properties` dtype to populate the `empty_properties` with arrays of the correct dtype.

To do this, we can change lines 288 in the snippet above to

```python
empty_properties = {k: np.empty(0, dtype=v.dtype) for k, v in properties.items()}
```

Constructing the layer now looks like:

```python
layer.add_points(
    properties={
        ‘class’: np.array([‘a’, ‘b’]),
        ‘value’: np.array([0], dtype=np.float
    }
)
```

Reply:
late to the party, but just curious whether it wouldn't be possible to call np.array() on whatever the user provides?  I definitely also like allowing the `Points(properties={'class': ['a', 'b']})` syntax if at all possible.

Reply:
@kevinyamauchi yeah, that works great for now! I still like `'value': float` as readable syntax, but we can leave that for a future PR.

I definitely agree with @tlambert03 that it would be great to not have to pass `np.array()`

Reply:
Sounds good. I've implemented it!